webmention.php

```diff
@@ -81,13 +114,34 @@ public static function parse_query($wp) {
     if (!array_key_exists('webmention', $wp->query_vars)) {
       return;
     }
-
-    $content = file_get_contents('php://input');
-    parse_str($content);
+    else {
+      // check if the end point has expired
+      $valid_ticks = array( 0, -1, -2 );
+      
+      $supplied_code = get_query_var( 'webmention' );
+      $is_valid = false;
+ 
+      foreach ( $valid_ticks as $tick ) {
+        if ( hash_equals( WebMentionPlugin::expire_code( $tick ), $supplied_code ) ) {
+          $is_valid = true;
+          break;
+        }
+      }
+    }
 
     // plain text header
     header('Content-Type: text/plain; charset=' . get_option('blog_charset'));
 
+    // fail if invalide endpoint
+    if ( false == $is_valid ) {
+      status_header(400);
```

Comment:
A `403` would probably be better here, http://httpstatus.es/403, as the server understands the request, but refuses to handle it.

A `400` is more of a generic error where the server don't really understand what the client actually wanted to do: http://httpstatus.es/400